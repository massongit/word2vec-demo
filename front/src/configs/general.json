{
  "word_embeddings.methods": {
    "word2vec": "Word2Vec",
    "fasttext": "FastText"
  }
}
