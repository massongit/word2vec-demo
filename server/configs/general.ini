;一般設定
;相対パスはserver/src/server.pyを基準として記述する

;分散表現関連の設定
[word embedding]
;使用する手法
;0: Word2Vec
;1: FastText
method =
;モデルのパス
;※FastTextのモデルを使用する場合はバイナリー形式のみ
model path =
;バイナリーかどうか (0 or 1)
;※FastTextのモデルを使用する場合、この項目は無視され、バイナリーとして扱われる
is binary =

;MeCab関連の設定
;※Word2Vecのモデルを使用する場合のみ
[mecab]
;Word2Vecのモデルを生成する際に使用したMeCab用の辞書のパス
dir path =

;ログ関連の設定
[log]
;ログを格納するディレクトリのパス
path = ../logs/

;フロントエンド関連の設定
[front]
;フロントエンドのビルド先のURL (同じ端末内にある場合は空欄)
url =
;フロントエンドのビルド先のディレクトリのパス
dir path = ../../front/build/

;サーバー関連の設定
;※server.pyを直接動かす場合のみ
;※uWSGIを使用する場合は無視される
[server]
;サーバーのホスト名 (デフォルトの設定の場合は空欄)
host =
;サーバーのポート (デフォルトの設定の場合は空欄)
port =
