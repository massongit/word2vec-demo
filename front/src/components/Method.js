import React from "react"
import PropTypes from "prop-types"
import general from "../configs/general"
import {Form, FormGroup, Input, Label} from "reactstrap"
import {intlShape} from "react-intl"
import {setMethod} from "../actions"

/**
 * 手法選択部
 */
class Method extends React.Component {
    static propTypes = {
        dispatch: PropTypes.func.isRequired,
        methods: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
        method: PropTypes.string,
        intl: intlShape.isRequired
    }

    /**
     * ラジオボタンを描画する
     * @returns {*} ラジオボタン
     */
    renderRadios() {
        return this.props.methods.map((m, i) => {
            if (m in general["word_embeddings.methods"]) {
                return (
                    <FormGroup
                        check
                        inline
                        key={i}
                    >
                        <Input
                            key={i}
                            type="radio"
                            name="method"
                            onChange={
                                () => {
                                    this.props.dispatch(setMethod({
                                        method: m
                                    }))
                                }
                            }
                        />
                        <Label check>
                            {general["word_embeddings.methods"][m]}
                        </Label>
                    </FormGroup>
                )
            } else {
                return undefined
            }
        })
    }

    /**
     * 手法の説明を描画する
     * @returns {*} 手法の説明
     */
    renderDescription() {
        if (this.props.method) {
            return this.props.intl.formatMessage({
                id: "description." + this.props.method
            }).split("\n").map((e, i) => (
                <p
                    key={i}
                    style={{
                        margin: 0
                    }}
                >
                    {e}
                </p>
            ))
        }
    }

    render() {
        return (
            <div>
                <Form inline>
                    {this.renderRadios()}
                </Form>
                {this.renderDescription()}
            </div>
        )
    }
}

export default Method
