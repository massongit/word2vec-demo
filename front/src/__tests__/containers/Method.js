import React from "react"
import Method from "../../containers/Method"
import thunk from "redux-thunk"
import configureMockStore from "redux-mock-store"
import general from "../../configs/general"
import setMethodsState from "../../test_data/setMethodsState"
import initialLoadingState from "../../test_data/initialLoadingState"
import initialSetMethodState from "../../test_data/initialSetMethodState"
import initialShowSimilarWordsState from "../../test_data/initialShowSimilarWordsState"
import {FormGroup, Radio} from "react-bootstrap"
import {loadTranslation, mountWithIntl, shallowWithIntl} from "enzyme-react-intl"

let store

loadTranslation("./src/translations/ja.json")

describe("containers/Method", () => {
    beforeEach(() => {
        store = configureMockStore([thunk])({
            setMethod: initialSetMethodState,
            setMethods: setMethodsState,
            loading: initialLoadingState,
            showSimilarWords: initialShowSimilarWordsState
        })
    })

    it("Componentが正しく配置されている", () => {
        const methodComponent = shallowWithIntl(
            <Method
                store={store}
            />
        ).dive()
        expect(methodComponent).toMatchSnapshot()
    })

    it("FormGroupになっている", () => {
        const methodComponent = mountWithIntl(
            <Method
                store={store}
            />
        )
        expect(methodComponent.contains(FormGroup)).toBeTruthy()
    })

    it("手法一覧セットActionのStateに含まれる全ての手法がRadioとして配置されている", () => {
        const methodComponent = mountWithIntl(
            <Method
                store={store}
            />
        )
        const radio = methodComponent.find(Radio)

        expect(radio).toHaveLength(setMethodsState.methods.length)

        for (const k of setMethodsState.methods) {
            expect(radio.contains(general["word_embeddings.methods"][k])).toBeTruthy()
        }
    })

    it("手法を選択したとき、正常に手法セットActionがdispatchされる", () => {
        const methodComponent = mountWithIntl(
            <Method
                store={store}
            />
        )
        methodComponent.find(Radio).forEach((k, i) => {
            k.props().onChange()
            expect(store.getActions()).toHaveLength(i + 1)
        })
    })
})
