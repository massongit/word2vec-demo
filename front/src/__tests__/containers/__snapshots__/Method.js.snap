// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Method Componentが正しく配置されている 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Method)
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "calculate": "計算",
          "description.fasttext": "2016年にFaceBook社によって提唱された手法であり、Word2Vecの改良版にあたります。
モデル内に存在しない単語のベクトルの推定や文書の分類ができるようになっています。",
          "description.input": "日本語の単語を空白で区切って入力し、計算ボタンをクリックしてください。",
          "description.input.supplement.1": "引き算を行う場合には、単語の先頭に『-』を付けます。",
          "description.input.supplement.2": "足し算を行う場合には、単語の先頭に『+』を付けることができます (基本的には付けなくても問題ありませんが、単語が『+』や『-』で始まる場合には必ず付けてください)。",
          "description.method": "以下から手法を選択してください。",
          "description.word2vec": "2013年にGoogle社によって提唱された手法であり、自然言語処理の様々な分野で用いられています。",
          "description.wordEmbedding": "単語を意味のベクトルで表現するための技術です。
これを利用して、単語の足し算や引き算を行うことができます。",
          "errorMessage.default": "エラーが発生しました: {message}",
          "errorMessage.predict": "計算中にエラーが発生しました: {message}",
          "header.candidate": "候補",
          "header.cosine": "コサイン類似度 (-1 〜 1)",
          "header.keywords": "<キーワード>",
          "header.word": "単語",
          "title.description": "概要",
          "title.input": "入力",
          "title.method": "手法",
          "title.result": "結果",
          "unknown": "(予測不能)",
        },
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "dispatch": [Function],
      "intl": Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "calculate": "計算",
          "description.fasttext": "2016年にFaceBook社によって提唱された手法であり、Word2Vecの改良版にあたります。
モデル内に存在しない単語のベクトルの推定や文書の分類ができるようになっています。",
          "description.input": "日本語の単語を空白で区切って入力し、計算ボタンをクリックしてください。",
          "description.input.supplement.1": "引き算を行う場合には、単語の先頭に『-』を付けます。",
          "description.input.supplement.2": "足し算を行う場合には、単語の先頭に『+』を付けることができます (基本的には付けなくても問題ありませんが、単語が『+』や『-』で始まる場合には必ず付けてください)。",
          "description.method": "以下から手法を選択してください。",
          "description.word2vec": "2013年にGoogle社によって提唱された手法であり、自然言語処理の様々な分野で用いられています。",
          "description.wordEmbedding": "単語を意味のベクトルで表現するための技術です。
これを利用して、単語の足し算や引き算を行うことができます。",
          "errorMessage.default": "エラーが発生しました: {message}",
          "errorMessage.predict": "計算中にエラーが発生しました: {message}",
          "header.candidate": "候補",
          "header.cosine": "コサイン類似度 (-1 〜 1)",
          "header.keywords": "<キーワード>",
          "header.word": "単語",
          "title.description": "概要",
          "title.input": "入力",
          "title.method": "手法",
          "title.result": "結果",
          "unknown": "(予測不能)",
        },
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      },
      "method": undefined,
      "methods": Array [
        "word2vec",
        "fasttext",
      ],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dispatch": [Function],
        "intl": Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {
            "calculate": "計算",
            "description.fasttext": "2016年にFaceBook社によって提唱された手法であり、Word2Vecの改良版にあたります。
モデル内に存在しない単語のベクトルの推定や文書の分類ができるようになっています。",
            "description.input": "日本語の単語を空白で区切って入力し、計算ボタンをクリックしてください。",
            "description.input.supplement.1": "引き算を行う場合には、単語の先頭に『-』を付けます。",
            "description.input.supplement.2": "足し算を行う場合には、単語の先頭に『+』を付けることができます (基本的には付けなくても問題ありませんが、単語が『+』や『-』で始まる場合には必ず付けてください)。",
            "description.method": "以下から手法を選択してください。",
            "description.word2vec": "2013年にGoogle社によって提唱された手法であり、自然言語処理の様々な分野で用いられています。",
            "description.wordEmbedding": "単語を意味のベクトルで表現するための技術です。
これを利用して、単語の足し算や引き算を行うことができます。",
            "errorMessage.default": "エラーが発生しました: {message}",
            "errorMessage.predict": "計算中にエラーが発生しました: {message}",
            "header.candidate": "候補",
            "header.cosine": "コサイン類似度 (-1 〜 1)",
            "header.keywords": "<キーワード>",
            "header.word": "単語",
            "title.description": "概要",
            "title.input": "入力",
            "title.method": "手法",
            "title.result": "結果",
            "unknown": "(予測不能)",
          },
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        },
        "method": undefined,
        "methods": Array [
          "word2vec",
          "fasttext",
        ],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "intl": Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "calculate": "計算",
          "description.fasttext": "2016年にFaceBook社によって提唱された手法であり、Word2Vecの改良版にあたります。
モデル内に存在しない単語のベクトルの推定や文書の分類ができるようになっています。",
          "description.input": "日本語の単語を空白で区切って入力し、計算ボタンをクリックしてください。",
          "description.input.supplement.1": "引き算を行う場合には、単語の先頭に『-』を付けます。",
          "description.input.supplement.2": "足し算を行う場合には、単語の先頭に『+』を付けることができます (基本的には付けなくても問題ありませんが、単語が『+』や『-』で始まる場合には必ず付けてください)。",
          "description.method": "以下から手法を選択してください。",
          "description.word2vec": "2013年にGoogle社によって提唱された手法であり、自然言語処理の様々な分野で用いられています。",
          "description.wordEmbedding": "単語を意味のベクトルで表現するための技術です。
これを利用して、単語の足し算や引き算を行うことができます。",
          "errorMessage.default": "エラーが発生しました: {message}",
          "errorMessage.predict": "計算中にエラーが発生しました: {message}",
          "header.candidate": "候補",
          "header.cosine": "コサイン類似度 (-1 〜 1)",
          "header.keywords": "<キーワード>",
          "header.word": "単語",
          "title.description": "概要",
          "title.input": "入力",
          "title.method": "手法",
          "title.result": "結果",
          "unknown": "(予測不能)",
        },
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      },
    },
  },
}
`;
